xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java

```diff
@@ -78,27 +77,33 @@ public PlainTextBlockFilter(Parser plainTextParser, LinkLabelGenerator linkLabel
     {
         if (VALID_PLAINTEXT_BLOCKS.contains(block.getClass())) {
             return Collections.singletonList(block);
-        } else if (LinkBlock.class.isAssignableFrom(block.getClass()) && block.getChildren().isEmpty()) {
+        }
+
+        if (LinkBlock.class.isAssignableFrom(block.getClass()) && block.getChildren().isEmpty()) {
             ResourceReference reference = ((LinkBlock) block).getReference();
+            return filterLinkBlock(reference);
+        }
 
-            try {
-                String label;
+        return Collections.emptyList();
```

Comment:
Are you not creating a lot of exit points for the method, which would incidentally generate a new issue? :)

Reply:
EDIT: probably not since I saw you split the method into 2.